train.py

```diff
@@ -127,6 +126,7 @@ def train_model(
             accum_iter=cfg["accum_iter"],
             log_frequency=cfg.log_frequency,
         )
+        del lr_scheduler, optimizer  # only deleting both lr_scheduler and optimizer frees memory
```

Comment:
I seem to have resolved memory leak issue by calling del on lr_scheduler and optimizer. I now initialize LambdaLR and LabelSmoothing objects in each epoch. What is interesting only deleting them both works, as deleting only one of them does not free the memory. Maybe you have an idea why that works?
I also use 
``` with torch.no_grad():
```
before evaluating which also prevents allocating unnecessary gradients on memory.

Reply:
I think it's not right to delete those objects here - optimizer has a state that should be maintained across epochs, so you should initialize it before `for epoch in range(epoch)` starts.

But if this helps, my next guess would be that the leak is connected to the gradient accumulation logic in the step function, where it looks like it's possible to return from that function without calling `.zero_grad()` if `accum_iter` does not split the number of batches evenly 🤔

Reply:
that makes sense